fragment/fragment/src/main/java/androidx/fragment/app/strictmode/FragmentStrictMode.java

```diff
@@ -0,0 +1,204 @@
+            @NonNull
+            @SuppressLint("BuilderSetStyle")
+            public Builder penaltyLog() {
+                flags.add(Flag.PENALTY_LOG);
+                return this;
+            }
+
+            /**
+             * Crash the whole process on violation. This penalty runs at the end of all enabled
+             * penalties so you'll still get to see logging or other violations before the process
+             * dies.
+             */
+            @NonNull
+            @SuppressLint("BuilderSetStyle")
+            public Builder penaltyDeath() {
+                flags.add(Flag.PENALTY_DEATH);
+                return this;
+            }
+
+            /** Call #{@link OnViolationListener#onViolation} for every violation. */
+            @NonNull
+            @SuppressLint("BuilderSetStyle")
+            public Builder penaltyListener(@NonNull OnViolationListener listener) {
+                this.listener = listener;
+                return this;
+            }
+
+            /**
+             * Construct the Policy instance.
+             *
+             * <p>Note: if no penalties are enabled before calling <code>build</code>, {@link
+             * #penaltyLog} is implicitly set.
+             */
+            @NonNull
+            public Policy build() {
+                if (listener == null && !flags.contains(Flag.PENALTY_DEATH)) {
+                    penaltyLog();
+                }
+                return new Policy(flags, listener);
+            }
+        }
+    }
+
+    /** Returns the current default policy. */
+     @NonNull
+    public static Policy getDefaultPolicy() {
+        return defaultPolicy;
+    }
+
+    /**
+     * Sets the policy for what actions should be detected, as well as the penalty if such actions
+     * occur.
+     *
+     * @param policy the policy to put into place
+     */
+    public static void setDefaultPolicy(@NonNull Policy policy) {
+        defaultPolicy = policy;
+    }
+
+    private static Policy getNearestPolicy(@Nullable Fragment fragment) {
+        while (fragment != null) {
+            if (fragment.isAdded()) {
+                FragmentManager fragmentManager = fragment.getParentFragmentManager();
+                if (fragmentManager.getStrictModePolicy() != null) {
+                    return fragmentManager.getStrictModePolicy();
+                }
+            }
+            fragment = fragment.getParentFragment();
+        }
+        return defaultPolicy;
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    static void onPolicyViolation(@NonNull Fragment fragment, @NonNull Violation violation) {
+        onPolicyViolation(getNearestPolicy(fragment), violation);
+    }
+
+    private static void onPolicyViolation(@NonNull Policy policy, @NonNull Violation violation) {
+        if (policy.flags.contains(Flag.PENALTY_LOG)) {
+            Log.d(TAG, "FragmentStrictMode policy violation: ", violation);
```

Comment:
FragmentStrictMode is already the tag, you can remove it from the string itself (and the trailing `: `) - just "Policy Violation". I think it would be very helpful to print out the causing Fragment with the log message though so that developers can tie this log message back to a particular fragment.